common/src/main/java/oracle/kubernetes/common/utils/SchemaConversionUtils.java

```diff
@@ -104,6 +120,128 @@ public String convertDomainSchema(String domainYaml) {
     return yaml.dump(convertDomainSchema((Map<String, Object>) domain));
   }
 
+  private void convertClustersToResources(ClusterCustomResourceHelper clusterCustomResourceHelper,
+      Map<String, Object> domain, String desiredAPIVersion) {
+    try {
+      if (clusterCustomResourceHelper != null) {
+        String apiVersion = (String)domain.get("apiVersion");
+        if (CommonConstants.API_VERSION_V8.equals(apiVersion)
+            && CommonConstants.API_VERSION_V9.equals(desiredAPIVersion)) {
+          convertV8toV9(clusterCustomResourceHelper, domain);
+        } else if (CommonConstants.API_VERSION_V9.equals(apiVersion)
+            && CommonConstants.API_VERSION_V8.equals(desiredAPIVersion)) {
+          convertV9toV8(clusterCustomResourceHelper, domain);
+        }
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  protected void convertV8toV9(ClusterCustomResourceHelper clusterCustomResourceHelper, Map<String, Object> domain)
+      throws ApiException {
+    Map<String, Object> domainSpec = getSpec(domain);
+    List<Object> clusterSpecs = getOrCreateClusterSpecs(domainSpec);
+    if (!clusterSpecs.isEmpty()) {
+      List<String> clusterResources = clusterCustomResourceHelper
+          .getNamesOfDeployedClusterResources(getNamespace(domain), getDomainUid(domain));
+      Iterator<Object> itr = clusterSpecs.iterator();
+      while (itr.hasNext()) {
+        Map<String, Object> clusterSpec = (Map<String, Object>) itr.next();
+        String clusterName = (String) clusterSpec.get("clusterName");
+        if (!clusterResources.contains(clusterName)) {
+          // Create Cluster Resource
+          clusterCustomResourceHelper.createClusterResource(clusterSpec, domain);
```

Comment:
There are likely several use cases here, such as where a Cluster resource already exists and is the same or different than the resource the code wants to create.